woocommerce/woocommerce-ios
# Update store creation status checking logic after the API fix

Closes: #9733 


## Why


Previously, a workaround was added to the app to wait for store creation completion by checking whether the Jetpack & WooCommerce plugins are installed and active on the site https://github.com/woocommerce/woocommerce-ios/pull/9731. Now that the timeout issue is fixed on the web side (more details in pe5sF9-1xN-p2#comment-2303), we no longer need to check the plugins endpoint. In this PR, the store creation completion status checking logic was updated following the plan in pe5sF9-1xN-p2#comment-2322:

- Always wait for `jetpack`/`jetpack_connection`/`woocommerce_is_active`/`is_wpcom_store` properties to turn `true` since they are critical in the app (`is_wpcom_store` is used for the free trial banner, settings domains, Jetpack AI, and maybe more)
- Increase the polling interval from 10s to 15s, and the max number of polling from 10 to 15 so that it’s a maximum ~3.75 minutes

## Testing instructions


- Log in with a WPCOM account if needed
- Go to the Menu tab, and tap on the store picker
- At the end of the store list, tap `Add a store`
- Continue the steps to create a free trial site --> after about 1-2 minutes, the flow should succeed and navigate to the newly created store with a free trial banner


---
